lib/proximity.js

```diff
@@ -190,6 +190,36 @@ var Controllers = {
       }
     }
   },
+  HCSR04I2C: {
+    initialize: {
+      value: function(opts, datahandler) {
+        var BYTES_TO_READ = 0x02;
+        var address = opts.address || 0x27;
+        var msToNextRead = 90;
+
+        opts.address = address;
+
+        // set up IO connection:
+        this.io.i2cConfig(opts);
+
+			  var read = function() {
+				  setTimeout(function() {
+					  this.board.io.i2cReadOnce(address, BYTES_TO_READ, function(data) {
+						  datahandler((data[0] << 8) + data[1]);
+              read();
+					  });
+				  }.bind(this), msToNextRead);
+			  }.bind(this);
+
+        read();
+      }
+    },
+    toCm: {
+      value: function(raw) {
+        return +(raw / 29.1 / 2).toFixed(3);
```

Comment:
Mostly because it can do sub cm detection so passing a float back is a bit grotty. In theory also you are supposed to account for air pressure when you calculate this so a raw value would allow you to accommodate for that (not sure anyone other than me has ever *actually* done that with a nodebot yet though ;P )